### PR TITLE
dynamically determine num_layers in _qwen3_hf_to_vllm

### PR DESCRIPTION
**Pull request by casteryh**
This is fixing #190

### CODE REVIEW

File: src/forge/actors/trainer.py [hunk 1]
```diff
@@ -215,7 +216,7 @@ async def push_weights(self, policy_version: int) -> None:
             )
         hf_state_dict = self.engine.checkpointer.sd_adapter.to_hf(flattened_state_dict)
         # TODO: Figure out how to gracefully handle which model to-vLLM conversion is needed
-        vllm_ready_hf_sd = _qwen3_hf_to_vllm(sd=hf_state_dict, num_layers=28)
+        vllm_ready_hf_sd = _qwen3_hf_to_vllm(sd=hf_state_dict)
```
Comment (Ritesh1905) [inline src/forge/actors/trainer.py:219]: Could this just be simplified by using `num_layers=self.model.config.num_hidden_layers`? see [this](https://github.com/meta-pytorch/forge/blob/main/apps/toy_rl/sumdigits.py#L344-L346)

Ideally you should not be needing this method in the trainer at all. The trainer should be agnostic to the type/arch of generator.

Comment (allenwang28) [reply src/forge/actors/trainer.py:219]: this should work with torchtitan:

```
num_layers=self.engine.model_args.n_layers)
```

Comment (casteryh) [reply src/forge/actors/trainer.py:219]: I guess simply reading the max num layers from regex has the advantage that it's agnostic of the trainer implementation as long as the state dict is in huggingface format.
Let me know what you think. @Ritesh1905 @allenwang28

**Comment by casteryh** [comment]
closed since the change is already in #215
